pgbouncer_custom/pgbouncer_custom.py

```diff
@@ -0,0 +1,148 @@
+import psycopg2 as pg
+import psycopg2.extras as pgextras
+from collections import defaultdict
+from six.moves.urllib.parse import urlparse
+
+from checks import AgentCheck
+
+
+class ShouldRestartException(Exception):
+    pass
+
+
+class PgBouncerCustom(AgentCheck):
+    # Adapted from https://github.com/DataDog/integrations-core/blob/master/pgbouncer/datadog_checks/pgbouncer/pgbouncer.py
+    DB_NAME = 'pgbouncer'
+    SERVICE_CHECK_NAME = 'pgbouncer.can_connect'
+    QUERY = 'SHOW CLIENTS'
+
+    def _get_service_checks_tags(self, host, port, database_url, tags=None):
+        if tags is None:
+            tags = []
+
+        if database_url:
+            parsed_url = urlparse(database_url)
+            host = parsed_url.hostname
+            port = parsed_url.port
+
+        service_checks_tags = [
+            "host:%s" % host,
+            "port:%s" % port,
+            "db:%s" % self.DB_NAME
+        ]
+        service_checks_tags.extend(tags)
+        service_checks_tags = list(set(service_checks_tags))
+
+        return service_checks_tags
+
+    def _collect_stats(self, db, instance_tags):
+        try:
+            with db.cursor(cursor_factory=pgextras.DictCursor) as cursor:
+                try:
+                    self.log.debug("Running query: %s", self.QUERY)
+                    cursor.execute(self.QUERY)
+                    rows = cursor.fetchall()
+                except pg.Error:
+                    self.log.exception("Not all metrics may be available")
+                else:
+                    count_by_db_and_host = defaultdict(lambda: defaultdict(int))
+                    for row in rows:
+                        self.log.debug("Processing row: %r", row)
+                        db_name = row[2]
+                        client_addr = row[4]
+                        count_by_db_and_host[client_addr][db_name] += 1
+                    for client, counts_by_host in count_by_db_and_host.iteritems():
+                        for db_name, count in counts_by_host.iteritems():
+                            tags = {'client_addr': client, 'db_name': db_name}
+                            tags.extend(instance_tags)
+                            self.gauge(
```

Comment:
Or we could just make a count for each connection and then we can get total counts in datadog, right?

Reply:
Will keep this as is.